0004-market-integration-token-standard.md

```diff
@@ -137,25 +157,28 @@ At time of writing, this standard is established with several constraints found
 
 `mintToken` - It's up to the developer of a token to decide how tokens are generated. Minting should return the Token with a unique ID attached.
 
-`transfer` - A straightforward transfer between two 
+`transfer` - A straightforward transfer between two
 
-needs collections
+## Collections
+```TypeScript
   tokens
   tokensByOwner
+```
 
 ## Models for AssemblyScript contract
 
 ```TypeScript
 export class Token {
-  id: string;
-  tokenType: TokenType;
-  data: bytes;
+  id:u64;
+  tokenTypeId:u32;
+  // serialized json
+  data:string;
```

Comment:
Is this data intended to be `readonly` ? 
If not, how would a consumer parse the data in a deterministic way to arrive at the same JSON as the original token.
It seems to be that having the data change would "promote" people using `data` as a way to store things like "level" or "value" of an NFT, leading to immutability issues.

Reply:
yes, it's expected that you want to store stuff like crypto-corgi DNA when you mint a token and then not change it

people can mess it up, but you really can mess up anything if not following convention while implementing token